### ISSUE TITLE
Classification example not running

### ISSUE THREAD

**Issue by Michael-Schulze** [EFS-OpenSource/calibration-framework#43]
I installed the calibration framework from scratch as described and was using an conda environment, but when I try to run the classification examples I got an error. It looks like that it is related to the ENIR.

Get path of all Near Isotonic Regression models with mPAVA ...
Traceback (most recent call last):
  File "/home/labor/calibration-framework/examples/classification/CIFAR.py", line 169, in <module>
    cross_validation(model, use_cuda=use_cuda, domain=domain)
  File "/home/labor/calibration-framework/examples/classification/CIFAR.py", line 135, in cross_validation
    success = cross_validation_5_2(models=models, datafile=datafile, bins=bins, save_models=save_models, domain=domain)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/Decorator.py", line 90, in new_f
    return f(*args, **kwds)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/examples/classification/utils.py", line 236, in cross_validation_5_2
    instance.fit(build_set_sm, build_set_gt)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/Decorator.py", line 62, in new_f
    return f(*args, **kwds)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/binning/ENIR.py", line 239, in fit
    self._multiclass_instances = self._create_one_vs_all_models(X, y, ENIR, self.score_function,
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/AbstractCalibration.py", line 568, in _create_one_vs_all_models
    model.fit(onevsall_confidence, onevsall_ground_truth)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/Decorator.py", line 62, in new_f
    return f(*args, **kwds)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/binning/ENIR.py", line 270, in fit
    self._model_scores, self._binning_models = self._elbow(X, y, model_list, self.score_function, alpha=0.001)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/Decorator.py", line 35, in new_f
    return f(*args, **kwds)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/AbstractCalibration.py", line 498, in _elbow
    model_scores = self._calc_model_scores(confidences, ground_truth, model_list, score_function)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/Decorator.py", line 35, in new_f
    return f(*args, **kwds)
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/netcal/AbstractCalibration.py", line 468, in _calc_model_scores
    model_scores = np.exp((np.min(score) - score) / 2.)
  File "<__array_function__ internals>", line 200, in amin
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/numpy/core/fromnumeric.py", line 2946, in amin
    return _wrapreduction(a, np.minimum, 'min', axis, None, out,
  File "/home/labor/miniconda3/envs/cal/lib/python3.10/site-packages/numpy/core/fromnumeric.py", line 86, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
ValueError: zero-size array to reduction operation minimum which has no identity

Used setup:
- Ubuntu 20.04

1) conda environment
`conda create --name cal python=3.10`
`conda activate cal`

2) Repo clone and install
`git clone https://github.com/EFS-OpenSource/calibration-framework`
`cd calibration-framework/`
`python3 -m pip install .`

3) Execute examples
`cd examples/classification/`
`python CIFAR.py`

**Comment by fabiankueppers** [comment]
Hi, sorry for my very late response ... did you already find a way to get it working? If so, please let me know. I will also take a look into the example in the meantime. Thank you.

**Comment by pallavimitra** [comment]
Hi,
I am facing the exact problem while trying to run the classification example CIFAR.py.


Pasting the error below:

Traceback (most recent call last):
  File "/home/uif62213/calibration-framework/examples/classification/CIFAR.py", line 169, in <module>
    cross_validation(model, use_cuda=use_cuda, domain=domain)
  File "/home/uif62213/calibration-framework/examples/classification/CIFAR.py", line 135, in cross_validation
    success = cross_validation_5_2(models=models, datafile=datafile, bins=bins, save_models=save_models, domain=domain)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/Decorator.py", line 90, in new_f
    return f(*args, **kwds)
           ^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/examples/classification/utils.py", line 236, in cross_validation_5_2
    instance.fit(build_set_sm, build_set_gt)
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/Decorator.py", line 62, in new_f
    return f(*args, **kwds)
           ^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/binning/ENIR.py", line 239, in fit
    self._multiclass_instances = self._create_one_vs_all_models(X, y, ENIR, self.score_function,
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/AbstractCalibration.py", line 568, in _create_one_vs_all_models
    model.fit(onevsall_confidence, onevsall_ground_truth)
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/Decorator.py", line 62, in new_f
    return f(*args, **kwds)
           ^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/binning/ENIR.py", line 270, in fit
    self._model_scores, self._binning_models = self._elbow(X, y, model_list, self.score_function, alpha=0.001)
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/Decorator.py", line 35, in new_f
    return f(*args, **kwds)
           ^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/AbstractCalibration.py", line 498, in _elbow
    model_scores = self._calc_model_scores(confidences, ground_truth, model_list, score_function)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/Decorator.py", line 35, in new_f
    return f(*args, **kwds)
           ^^^^^^^^^^^^^^^^
  File "/home/uif62213/.local/lib/python3.11/site-packages/netcal/AbstractCalibration.py", line 468, in _calc_model_scores
    model_scores = np.exp((np.min(score) - score) / 2.)
                           ^^^^^^^^^^^^^
  File "/opt/miniconda/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 2953, in min
    return _wrapreduction(a, np.minimum, 'min', axis, None, out,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/miniconda/lib/python3.11/site-packages/numpy/core/fromnumeric.py", line 88, in _wrapreduction
    return ufunc.reduce(obj, axis, dtype, out, **passkwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: zero-size array to reduction operation minimum which has no identity

**Comment by Michael-Schulze** [comment]
Below is the last stored env.yaml. As far as I remember, I just skipped the ENIR and used just some parts of the cal framework.
[environment.txt](https://github.com/user-attachments/files/16548449/environment.txt)

**Comment by fabiankueppers** [comment]
I was able to reproduce this issue. It seems that this is a deeper issue within the mPAVA algorithm of the NearIsotonicRegression class which is ENIR is based on. However, it seems that a deeper investigation is necessary which I'm currently not capable of. Since this issue only occured on the data for the "wideresnet" example, I excluded it from the example script to get at least the example work. I close this issue for now and mark it for the future.
